service/src/main/java/bio/terra/workspace/service/resource/model/WsmResourceApiFields.java

```diff
@@ -0,0 +1,23 @@
+package bio.terra.workspace.service.resource.model;
+
+import bio.terra.workspace.common.logging.model.ActivityLogChangeDetails;
+import bio.terra.workspace.service.logging.WorkspaceActivityLogService;
+import java.time.OffsetDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Api fields in WsmResource that is not stored as common fields in the WsmResource and needs to be
+ * fetched elsewhere.
+ */
+public record WsmResourceApiFields(String lastUpdatedBy, OffsetDateTime lastUpdatedDate) {
```

Comment:
This is awkward. I am thinking we should integrate this into the resource hierarchy rather than have to do this extra action all over the code.

We have a general problem with information that is not stored in the resource row, but is found elsewhere (activity log, cloud platform). I'd like to think of a more general solution than this.

Reply:
i just learned that I *can* have a final setter to autowire the ActivityLogservice to `WsmResource`. wdyt?

toApiResource seems OK to me because it's available for all the resource and it's the gateway to convert wsm resource to the api wsm resources. Any alternative that involve computing the non-resource common fields separately is not reliable.

Reply:
nvm. I can't inject it into `WsmResource`. Neither can I do it in the `WsmResourceHandler` i dont' think